### DISCUSSION TITLE
Build fails when processing a node_modules package(has es6 + jsx) with Parcel2

### DISCUSSION THREAD

**Discussion by rr-tommy-tan** [parcel-bundler/parcel#5412] [Q&A]
I'm trying to process a package in node_modules since it's written in es6 and has tsx (my project also uses tsx).  I've created a `babel.config.json` file in my project root which looks like this:
```
{
  "include": [
    "node_modules/@packageToProcessWithParcel/**/*",
    "path/to/project/source/files/**/*"
  ],
  "plugins": [
    "@babel/plugin-transform-react-jsx",
    ["@babel/plugin-proposal-class-properties", { "loose": true }]
  ],
  "presets": [
    "@babel/preset-env",
    "@babel/preset-typescript",
    "@babel/preset-react"
  ]
}
```
When running `watch` parcel seems to be able to process all my source files, but once it reaches the files from the `node_modules` package, the build fails and says:
```
🚨 Build failed.
@parcel/transformer-js: script' (19:7
SyntaxError: script' (19:7
  18 | export * from '@chakra-ui/layout';
> 19 | export type { GridProps, ListProps } from '@chakra-ui/layout';
>    |       ^
  20 | export * from '@chakra-ui/media-query';
  21 | export * from '@chakra-ui/theme';
```
Is there something I'm missing to be able to process the package using the parcel/babel-config from my own project?

relevant package.json:
```
    "@babel/core": "^7.12.9",
    "@babel/plugin-proposal-class-properties": "^7.12.1",
    "@babel/plugin-transform-react-jsx": "^7.12.7",
    "@babel/preset-env": "^7.12.7",
    "@babel/preset-react": "^7.12.7",
    "@babel/preset-typescript": "^7.12.7",
    "babel-plugin-transform-remove-console": "^6.9.4",
    "parcel": "^2.0.0-beta.1",
    "typescript": "^4.0.2",
```

**Comment by mischnic** [comment]
Is this a package from npm?

**Reply by rr-tommy-tan** [reply]
the package is not from npm.  it is from a private registry

**Comment by rr-tommy-tan** [comment]
i just found in the docs that there's a `includeNodeModules` (https://v2.parceljs.org/configuration/package-json/#includenodemodules).  will try this out now.

**Reply by rr-tommy-tan** [reply]
> i just found in the docs that there's a `includeNodeModules` (https://v2.parceljs.org/configuration/package-json/#includenodemodules). will try this out now.

adding this property into `package.json` didn't seem to work.  the same error posted above still occurs.
```
  "includeNodeModules": {
    "@packageScope/packageName": true
  },
```
also tried the array way of writing it but same result

@mischnic am i misunderstanding how this property should work? i was expecting the node_modules package to be processed as if it was like source code.

**Reply by mischnic** [reply]
By default, if you're building for Node, node modules are excluded. `includeNodeModules` can be used to enable that explcitily (or to exclude specific packages when building a library).

> i was expecting the node_modules package to be processed as if it was like source code.

That isn't possible at the moment. Ideally, ` "include": [    "node_modules/@packageToProcessWithParcel/**/*",` would work, but this isn't the case the moment.

**Reply by rr-tommy-tan** [reply]
> By default, if you're building for Node, node modules are excluded. `includeNodeModules` can be used to enable that explcitily (or to exclude specific packages when building a library).
> 
> > i was expecting the node_modules package to be processed as if it was like source code.
> 
> That isn't possible at the moment. Ideally, ` "include": [ "node_modules/@packageToProcessWithParcel/**/*",` would work, but this isn't the case the moment.

oh i see.  so just to clarify, `includeNodeModules` would work if we're building a library and building for `Node` and since I'm just building an app, this won't work?

Also, using `include` in `babel.config.json` is the right way to go, but atm there might be a bug?

**Reply by mischnic** [reply]
> includeNodeModules would work if we're building a library and building for Node and since I'm just building an app, this won't work?

No, if you're building for Node or libraries, external packages aren't bundled (so you will have things like `require("lodash")` in your output) by default.

> include in babel.config.json is the right way to go, but atm there might be a bug?

"If we do support explicitly transpiling node_modules at some point, it would most likely be via that option".

**Reply by rr-tommy-tan** [reply]
> > includeNodeModules would work if we're building a library and building for Node and since I'm just building an app, this won't work?
> 
> No, if you're building for Node or libraries, external packages aren't bundled (so you will have things like `require("lodash")` in your output) by default.
> 
> > include in babel.config.json is the right way to go, but atm there might be a bug?
> 
> "If we do support explicitly transpiling node_modules at some point, it would most likely be via that option".

oh ok.  so there is no way to transpile any package in node_modules? is there anything you can recommend?
